### ISSUE TITLE
RPC endpoints: compatibility tests are not running

### ISSUE THREAD

**Issue by Liamsi** [interchainio/tendermint-rs#41]
I was looking into the RPC endpoints and noticed that the integration tests are not running anymore. 

I think that is because the [`tests`](https://github.com/interchainio/tendermint-rs/tree/master/tests) directory got moved out of the module and is now a separate directory. Moving them back into a module (e.g. the `tendermint` directory fixes this). Alternatively, `tests` could be a separate crate referred to by the root Cargo.toml.

(related #38 #4)

**Comment by tarcieri** [comment]
@Liamsi the RPC integration tests that require a live `gaiad` instance are all marked as `#[ignore]` as a live `gaiad` instance may or may not be available. See notes here:

https://github.com/interchainio/tendermint-rs/blob/master/tests/integration.rs

tl;dr:

```
cargo test -- --ignored
```

**Comment by Liamsi** [comment]
Yes, noticed that too. But also the non-ignored fixture-based tests do not run; e.g. https://github.com/interchainio/tendermint-rs/blob/06eab13d173e51c20464104e8c6cbe6b4cffe2f9/tests/rpc.rs#L161-L166

Or is this also on purpose?

**Comment by tarcieri** [comment]
Aah, yes you're right, `tests` should get moved to `tendermint/tests`

**Comment by Liamsi** [comment]
OK, thanks! Will include that in the PR and will push in a sec. 

We could instead also make `tests` a separate crate. Similarly to what prost does here:
https://github.com/danburkert/prost/tree/master/tests
and
https://github.com/danburkert/prost/blob/9551f2852e414df72339634087c46ce5a08e85b2/Cargo.toml#L26
But we can discuss this separately or do so when `tests` grows larger.

**Comment by Liamsi** [comment]
Resolved via #42
